db.go

```diff
@@ -282,6 +285,17 @@ type Result interface {
 	// Where() discards the initial filtering conditions and sets new ones.
 	Where(...interface{}) Result
 
+	Join(...interface{}) Result
```

Comment:
How come you have Join, LeftJoin, FullJoin, etc... on the Result interface? why not keep the Result the same and have just the Builder() have those methods? for example, similar to how upper.io/db v2 already works with squirrel in the Pressly API. Keeps builder separated from basic result set operations.